pandas/core/series.py

```diff
@@ -3002,7 +3002,7 @@ def autocorr(self, lag: int = 1) -> float:
         >>> s.autocorr()
         nan
         """
-        return self.corr(self.shift(lag))
+        return self.corr(cast(Series, self.shift(lag)))
```

Comment:
This is necessary because Series.shift can now return a dataframe if there are multiple periods. Mypy complains at this line because it wants Series and not `Series | Dataframe`, but because there is only one lag, it's always `Series`, so we can just `cast`.

Reply:
You could add type-overloads to handle this. Sometimes they can get gnarly (see e.g. concat), but I think this one should be relatively straight forward.

Reply:
This is outdated - some recent commit I merged in from master made this unnecessary and I removed all of these. (Or maybe some changes I made)

Reply:
I'd prefer not to if that's ok? There is no `Series.shift` to overload so I feel like it would be confusing to newcomers like me.

Reply:
If you insist I will try!

Reply:
Otherwise I think this is done?

Reply:
Hi @rhshadrach , do you have time for another round/sign this off?

Reply:
I'm okay with opening an issue regarding this once it's merged for someone to followup on.